### PR TITLE
Implement Auth

### PR DESCRIPTION
**Pull request by Podgore**

### CODE REVIEW

File: StudyHub/StudyHub.DAL/Repositories/UserRepository.cs
Comment (fedyafedko) [inline StudyHub/StudyHub.DAL/Repositories/UserRepository.cs:1]: Delete UserRepository and use UserManager. URL:https://metanit.com/sharp/aspnet5/16.11.php

File: StudyHub/StudyHub.Common/DTO/LoginUserDTO.cs
Comment (fedyafedko) [inline StudyHub/StudyHub.Common/DTO/LoginUserDTO.cs:1]: fix all warnings

File: StudyHub/StudyHub/appsettings.json
Comment (fedyafedko) [inline StudyHub/StudyHub/appsettings.json:1]: Return the connection string and create another local json file and add your connection string there. Also leave the JwtSettings here and add to the local json file

Comment (aquaprogit) [reply StudyHub/StudyHub/appsettings.json:1]: Fedya meant adding local appsettings.development.json and using it locally, and common appsettings.json should contain shared data or placeholders like connection_string

File: StudyHub/StudyHub.DAL/EF/ApplicationDbContextFactory.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ class ApplicationDbContextFactory : IDesignTimeDbContextFactory<ApplicationDbCon
 {
     public ApplicationDbContext CreateDbContext(string[] args)
     {
-        var connectionString = "Server=DESKTOP-9SD4OV1;Database=StudyHub;Trusted_Connection=True;TrustServerCertificate=True";
+        var connectionString = "Server=(local);Database=StudyHub;Trusted_Connection=True;TrustServerCertificate=true;";
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/EF/ApplicationDbContextFactory.cs:9]: Maybe try to get this from appsettings excluding literal directly from file

File: StudyHub/StudyHub.DAL/Repositories/Repository.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ public class Repository<TEntity> : IRepository<TEntity>
     where TEntity : class
 {
     private readonly ApplicationDbContext _context;
-    private readonly DbSet<TEntity> _table;
+    public DbSet<TEntity> _table;
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/Repositories/Repository.cs:13]: No need to make it public

**Review by aquaprogit — CHANGES_REQUESTED** [review]

File: StudyHub/StudyHub/Controllers/AuthController.cs
Comment (fedyafedko) [inline StudyHub/StudyHub/Controllers/AuthController.cs:1]: Add try catch for controllers

File: StudyHub/StudyHub.BLL/Services/AuthService.cs
Comment (fedyafedko) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:1]: add refresh token

File: StudyHub/StudyHub.BLL/AutoMapper/UserProfile.cs
Comment (fedyafedko) [inline StudyHub/StudyHub.BLL/AutoMapper/UserProfile.cs:1]: why a profile if you didn't use a mapper ?

File: StudyHub/StudyHub.BLL/Services/AuthService.cs
Comment (Podgore) [reply StudyHub/StudyHub.BLL/Services/AuthService.cs:1]: It need to be done by another task, I'll register it

**Review by fedyafedko — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -397,3 +397,4 @@ FodyWeavers.xsd
 # JetBrains Rider
 *.sln.iml
 /StudyHub/StudyHub/appsettings.Development.json
+/StudyHub/StudyHub/appsettings.develop.json
```
Comment (aquaprogit) [inline .gitignore:400]: need to be removed and rename your local appsettings to **appsettings.Development.json**

File: StudyHub/Entities/StudyHub.Entities.csproj [hunk 4]
```diff
@@ -7,6 +7,8 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="AutoMapper" Version="12.0.1" />
```
Comment (aquaprogit) [inline StudyHub/Entities/StudyHub.Entities.csproj:10]: Is AutoMapper needed in this csproj?

File: StudyHub/Entities/Subject.cs [hunk 5]
```diff
@@ -4,8 +4,8 @@
 namespace StudyHub.Entities;
 public class Subject : EntityBase
 {
-    [ForeignKey(nameof(Teacher))]
+    [ForeignKey(nameof(User))]
     public Guid TeacherId { get; set; }
     public string Title { get; set; } = string.Empty;
-    public Teacher Teacher { get; set; } = null!;
+    public User User { get; set; } = null!;
```
Comment (aquaprogit) [inline StudyHub/Entities/Subject.cs:10]: Why used User here?

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+        if(!result)
+            throw new UnauthorizedAccessException(user.Email);
+
+        return new AuthSuccessDTO(GenerateJwtToken(existingUser));
+    }
+
+    public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO user)
+    {
+
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:42]: empty line should be removed

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        var existingUser = await _userManager.FindByEmailAsync(user.Email!);
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:43]: why user.Email can be null?

File: StudyHub/StudyHub.Common/DTO/LoginUserDTO.cs
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/DTO/LoginUserDTO.cs:1]: Why nullable props for mandatory fields?

File: StudyHub/StudyHub.Common/DTO/RegisterUserDTO.cs
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/DTO/RegisterUserDTO.cs:1]: Same. How is it possible to register with all null?

File: StudyHub/StudyHub.Common/StudyHub.Common.csproj [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="AutoMapper" Version="12.0.1" />
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/StudyHub.Common.csproj:10]: Is AutoMapper needed in this csproj?

File: StudyHub/StudyHub.DAL/EF/ApplicationDbContext.cs
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/EF/ApplicationDbContext.cs:1]: Discuss which method you shall use in your project, Fluent builder or Attribute usage (_currently both, unacceptable_)

File: StudyHub/StudyHub.DAL/StudyHub.DAL.csproj [hunk 9]
```diff
@@ -7,6 +7,8 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="AutoMapper" Version="12.0.1" />
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/StudyHub.DAL.csproj:10]: Is AutoMapper needed in this csproj?

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by fedyafedko — APPROVED** [review]

**Pull request merged**